README.md

```diff
@@ -387,6 +391,45 @@ module.exports = {
 };
 ```
 
+#### `AsyncFunction`
+
+You can set the `preprocessor` option as an Asynchronous Function (`AsyncFunction`) instance.
+
+**webpack.config.js**
+
+```js
+const Handlebars = require('handlebars');
+
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.hbs$/i,
+        loader: 'html-loader',
+        options: {
+          preprocessor: async (content, loaderContext) => {
+            let result;
+
+            try {
+              result = await Handlebars.compile(content)({
+                firstname: 'Value',
+                lastname: 'OtherValue',
+              });
+            } catch (error) {
+              await loaderContext.emitError(error);
+
+              return content;
+            }
+
+            return result;
+          },
+        },
+      },
+    ],
+  },
+};
+```
+
```

Comment:
No need `AsyncFunction` header, just add somehintg like that: `You can use async function, for example:`